File: Array Operations[12-May-2023]/solution.cpp
//  >> C++ CODE <<


int arrayOperations(int n, vector<int> &arr) {
        int cnt=0;
        bool flag=false,allNonZero=true;
        
        for(int i=0;i<n;i++)
        {
            if(arr[i]!=0)
            {
                if(flag==false) cnt++;
                flag=true;
            }
            else
            {
                allNonZero=false;
                flag=false;
            }
        }
        
        if(allNonZero) return -1;
        
            return cnt;
    }

File: Array Operations[12-May-2023]/solution.py
#  >> PYTHON CODE < 


class Solution:
    def arrayOperations(self, n : int, arr : List[int]) -> int:
    
        cnt=0
        flag=False
        allNonZero=True
        for i in range(n):
            if(arr[i]!=0):
                if(not flag): cnt+=1
                flag=True
            else:
                allNonZero=False
                flag=False
        
        if(allNonZero):
            return -1
            
        return cnt
